phys2bids/io.py

```diff
@@ -209,46 +122,55 @@ def process_acq(channel_list, chtrig, header=[]):
         elif freq_unit == 'kHz':
             freq = freq * 1000
         interval[0] = 1 / freq
-        freq = [freq] * (len(timeseries) + 1)
+        freq = [freq] * len(timeseries)
     else:
         # check if interval is in seconds, if not change the units to seconds and
         # calculate frequency
-        if interval[-1].split('/')[0] != 'sec':
-            LGR.warning('Interval is not in seconds. Converting its value.')
-            if interval[-1].split('/')[0] == 'min':
+        if interval[-1] != 's':
+            LGR.warning('Interval is not in seconds. Converting its value and unit.')
+            if interval[-1] == 'min':
                 interval[0] = float(interval[0]) * 60
                 interval[-1] = 's'
-            elif interval[-1].split('/')[0] == 'msec':
+            elif interval[-1] == 'msec':
                 interval[0] = float(interval[0]) / 1000
                 interval[-1] = 's'
-            elif interval[-1].split('/')[0] == 'µsec':
+            elif interval[-1] == 'µsec':
                 interval[0] = float(interval[0]) / 1000000
                 interval[-1] = 's'
+            if interval[-1] == 'hr':
+                interval[0] = float(interval[0]) * 3600
+                interval[-1] = 's'
+            elif interval[-1] == 'ms':
+                interval[0] = float(interval[0]) / 1000
+                interval[-1] = 's'
+            elif interval[-1] == 'µs':
+                interval[0] = float(interval[0]) / 1000000
+                interval[-1] = 's'
+            elif interval[-1] == 'sec':
```

Comment:
I'm not totally sure that this test is necessary any more, given that there is no use for `interval[-1]` after here. In fact, you could even remove all the `interval[-1] = 's' in the elif statements!

Reply:
It is necessary, remember that ``interval=["400", "Hz"]`` . Therefore we need to access only the second item, which is the one containing the unit information

Reply:
Let me try again.
The last moment you use `interval[-1]` in this function is during this `if` statement, to see if the value of `interval[0]` should be "corrected". After this `if` statement, `interval[-1]` becomes unused - from what I see, you're adding the unit `s` in the `unit` list in the hard way (coherently with your script).
For this reason, every `interval[-1]='s'` in this `elif` statement can be removed (unless you're keeping it for testing, but then please write it).
Lines 152-154 are actually superfluous. Line 152 and 154 do exactly what `phys2bids.bids.bidsify_unit()` does downstream, and line 153 is the same as line 156. The only change you would have to do (that is not wrong) is adding the option in line 131 for `interval[-1] != 's' or interval[-1] != 'sec'`. Which also makes sense for the message you're giving after in 132. It would be weird, if I know that I set up my labchart to record my sampling interval in seconds, to see that the sampling interval is not second.

You can save almost ten lines, I would do it!